worlds/lingo/__init__.py

```diff
@@ -0,0 +1,216 @@
+from .items import LingoItem, StaticLingoItems
+from .locations import LingoLocation, StaticLingoLocations
+from .Options import lingo_options, get_option_value
+from .testing import LingoTestOptions
+from ..generic.Rules import set_rule
+from .rules import LingoLogic, set_rules
+from .player_logic import LingoPlayerLogic
+from math import floor
+
+
+class LingoWebWorld(WebWorld):
+    theme = "grass"
+    tutorials = [Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to playing Lingo with Archipelago.",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["hatkirby"]
+    )]
+
+
+class LingoWorld(World):
+    """
+    Lingo is a first person indie puzzle game in the vein of The Witness. You find yourself in a mazelike, non-Euclidean
+    world filled with 800 word puzzles that use a variety of different mechanics.
+    """
+    game = "Lingo"
+    web = LingoWebWorld()
+
+    base_id = 444400
+
+    static_logic = StaticLingoLogic()
+    static_items = StaticLingoItems(base_id)
+    static_locat = StaticLingoLocations(base_id)
+    option_definitions = lingo_options
+
+    item_name_to_id = {
+        name: data.code for name, data in static_items.ALL_ITEM_TABLE.items()
+    }
+    location_name_to_id = {
+        name: data.code for name, data in static_locat.ALL_LOCATION_TABLE.items()
+        if data.code is not None
+    }
+
+    # This is just used for unit testing. It should remain at the default values for actual play.
+    test_options: LingoTestOptions = LingoTestOptions()
+
+    def _get_slot_data(self):
+        return {
+            'door_ids_by_item_id': {
+                data.code: data.door_ids for name, data in self.static_items.ALL_ITEM_TABLE.items()
+                if data.code is not None and len(data.door_ids) > 0
+            },
+            'painting_ids_by_item_id': {
+                data.code: data.painting_ids for name, data in self.static_items.ALL_ITEM_TABLE.items()
+                if data.code is not None and len(data.painting_ids) > 0
+            },
+            'panel_ids_by_location_id': {
+                data.code: data.panel_ids() for name, data in self.static_locat.ALL_LOCATION_TABLE.items()
+                if data.code is not None
+            },
+            'seed': self.multiworld.per_slot_randoms[self.player].randint(0, 1000000),
+        }
+
+    def generate_early(self):
+        self.player_logic = LingoPlayerLogic(self.multiworld, self.player, self.static_logic, self.test_options)
+        self.topology_present = get_option_value(self.multiworld, self.player, "shuffle_paintings")
+
+    def create_region(self, room: Room):
+        new_region = Region(room.name, self.player, self.multiworld)
+
+        if room.name in self.player_logic.LOCATIONS_BY_ROOM.keys():
+            for location in self.player_logic.LOCATIONS_BY_ROOM[room.name]:
+                new_loc = LingoLocation(self.player, location.name, location.code, new_region)
```

Comment:
Locations and regions should be made inside create_regions

Reply:
I broke the region logic out into a separate file, which I hope makes it clearer that it's all being done inside create_regions.